pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -190,6 +193,45 @@ const (
 	KernelTunableModify KernelTunableBehavior = "modify"
 )
 
+// Init creates the top level qos cgroup containers
+// We currently create containers for the Burstable and Best Effort containers
+// All guaranteed pods are nested under the RootContainer by default
+// Init is called only once during kubelet bootstrapping.
+func InitQOS(qosConfig *QOSConfig) (QOSContainersInfo, error) {
+	cm := NewCgroupManagerImpl()
+	// The rootContainer under which all qos containers are brought up is configurable
+	// and can be specified through the --cgroup-root flag.
+	// We default to the system root / in case no root is specified
+	rootContainerName := defaultContainerRoot
```

Comment:
In that case, let's add the default via the flag, to make it explicit to
the user.

On Thu, Jun 23, 2016 at 4:28 PM, Buddha Prakash <notifications@github.com>
wrote:

> In pkg/kubelet/cm/container_manager_linux.go
> <https://github.com/kubernetes/kubernetes/pull/27853#discussion_r68332856>
> :
>
> > @@ -190,6 +193,45 @@ const (
> >  	KernelTunableModify KernelTunableBehavior = "modify"
> >  )
> >
> > +// Init creates the top level qos cgroup containers
> > +// We currently create containers for the Burstable and Best Effort containers
> > +// All guaranteed pods are nested under the RootContainer by default
> > +// Init is called only once during kubelet bootstrapping.
> > +func InitQOS(qosConfig *QOSConfig) (QOSContainersInfo, error) {
> > +	cm := NewCgroupManagerImpl()
> > +	// The rootContainer under which all qos containers are brought up is configurable
> > +	// and can be specified through the --cgroup-root flag.
> > +	// We default to the system root / in case no root is specified
> > +	rootContainerName := defaultContainerRoot
>
> I believe currently cgroup root defaults to "". and not "/" hence the
> default.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27853/files/773b7e7144cb41cbb65d45b4e902e3888636af17..274b013b331ab329a043a235d717b1d3d578811b#r68332856>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGvIKNYuqpRpf78z7IkYcMEmh3aUecJMks5qOxaCgaJpZM4I7o6F>
> .
>